subsys/sdfw_services/services/device_info/device_info_service.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** .. include_startingpoint_device_info_source_rst */
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <sdfw/sdfw_services/device_info_service.h>
+
+#include "device_info_service_decode.h"
+#include "device_info_service_encode.h"
+#include "device_info_service_types.h"
+#include <sdfw/sdfw_services/ssf_client.h>
+#include "ssf_client_os.h"
+
+#define UUID_BYTES_LENGTH   16UL
+
+SSF_CLIENT_LOG_REGISTER(device_info_service, CONFIG_SSF_DEVICE_INFO_SERVICE_LOG_LEVEL);
+
+SSF_CLIENT_SERVICE_DEFINE(device_info_srvc, DEVICE_INFO, cbor_encode_device_info_req,
+			  cbor_decode_device_info_resp);
+
+
+int ssf_device_info_get_uuid(uint32_t *uuid_words, size_t uuid_words_count)
+{
+	int err = -1; /* initialize with negative value for error */
+
+	if ((NULL != uuid_words) && (uuid_words_count >= UUID_BYTES_LENGTH)) {
```

Comment:
Is there some reason for having the bigger than `>=` instead of equals `==`?` I'd assume UUID is always fixed in length.

Reply:
the read buffer and its size are provided by the user, it is okay to provide bigger that necessary buffer. In any case, the API will ALWAYS copy only 16 bytes to the output since, as you mentioned, the UUID has a fixed size.